bot/worker.py

```diff
@@ -65,116 +74,290 @@ def process_replyable(reddit, replyable):
+    * Comment beings with "try"
+    * Comment ends with valid hero name
+    * Comment was made as a reply to bot's comment
+    * Given hero has the original response
+    * Comment was added by OP, who made the original request for the response.
+    * Root/Original comment/submission was not hero specific response.
+
+    The comment tree would look something like below, where root(original) replyable can be Comment or Submission.
+    Only valid case is c3.
+    c1/s1 user: Foo
+        c2 bot: "Foo" response by Bar hero
+           c3 user: Try Bar2
+           c4 other_user: Try Bar2
 
+    :param reddit: The reddit account instance
     :param replyable: The comment/submission on reddit
-    :param processed_text: The processed body text
+    :param text: The processed body/title text
+    :return: True if this is a valid request, else False
+    """
+    # TODO confirm this keyword
+    if not text.startswith('try '):
+        return False
+
+    hero_name = text.split('try ')[1].strip()
+    hero_id = db_api.get_hero_id_by_name(hero_name=hero_name)
+    if hero_id is None:
+        return False
+
+    if not isinstance(replyable, Comment):
+        return False
+
+    op = replyable.author
+    parent_comment = replyable.parent()
+
+    if not isinstance(parent_comment, Comment):
+        return False
+
+    if not parent_comment.author == reddit.user.me():
+        return False
+
+    root_replyable = parent_comment.parent()
+
+    if not root_replyable.author == op:
+        return False
+
+    processed_text = process_text(root_replyable.body if isinstance(root_replyable, Comment) else root_replyable.title)
+
+    if is_hero_specific_response(processed_text):
+        return False
+
+    link, _ = db_api.get_link_for_response(processed_text=processed_text, hero_id=hero_id)
+
+    return link is not None
+
+
+def update_reply(replyable, text):
+    """Method to edit and update existing response comment by the bot with a new hero as requested.
+
+    :param replyable: The comment/submission on reddit
+    :param text: The processed body/title text
     :return: None
     """
+    bot_comment = replyable.parent()
+    root_replyable = bot_comment.parent()
 
-    link, hero_id = db_api.get_link_for_response(processed_text=processed_text)
+    # TODO maybe get original text from bot's command, rather than the original post, as it might be edited by the time this command is called
+    original_text = root_replyable.body if isinstance(root_replyable, Comment) else root_replyable.title
+    original_text = original_text.strip()
 
-    if link and hero_id:
-        img_dir = db_api.get_img_dir_by_id(hero_id=hero_id)
+    if '>' in original_text:
+        original_text = get_quoted_text(original_text).strip()
 
-        replyable.reply(create_reply(replyable=replyable, response_url=link, hero_id=hero_id, img=img_dir))
+    hero_name = text.split('try ')[1].strip()
+    hero_id = db_api.get_hero_id_by_name(hero_name=hero_name)
```

Comment:
It might add more complexity, as I would have to parse the bot's comment to get the hero's name, but it shouldn't be difficult or costly to do. Let me take this up in the next update.